netebpfext/net_ebpf_ext_prog_info_provider.c

```diff
@@ -13,17 +13,23 @@
  */
 typedef struct _net_ebpf_extension_program_info_client
 {
-    HANDLE nmr_binding_handle; ///< NMR binding handle.
-    GUID client_module_id;     ///< NMR module Id.
+    HANDLE nmr_binding_handle;                                    ///< NMR binding handle.
+    GUID client_module_id;                                        ///< NMR module ID.
+    ebpf_get_program_context_t get_program_context;               ///< Pointer to function to get the program context.
+    net_ebpf_extension_program_info_provider_t* provider_context; ///< Pointer to the hook NPI provider context.
 } net_ebpf_extension_program_info_client_t;
 
 /**
  *  @brief This is the program information NPI provider.
  */
 typedef struct _net_ebpf_extension_program_info_provider
 {
-    NPI_PROVIDER_CHARACTERISTICS characteristics; ///< NPI Provider characteristics.
-    HANDLE nmr_provider_handle;                   ///< NMR binding handle.
+    NPI_PROVIDER_CHARACTERISTICS characteristics;                     ///< NPI Provider characteristics.
+    HANDLE nmr_provider_handle;                                       ///< NMR binding handle.
+    net_ebpf_extension_program_info_on_client_attach attach_callback; /*!< Pointer to program info provider specific
+                                                              callback to be invoked when a client attaches. */
+    net_ebpf_extension_program_info_on_client_detach detach_callback; /*!< Pointer to program info provider specific
+                                                              callback to be invoked when a client detaches. */
```

Comment:
```suggestion
    net_ebpf_extension_program_info_on_client_attach attach_callback; ///< Pointer to program info provider specific
                                                              callback to be invoked when a client attaches.
    net_ebpf_extension_program_info_on_client_detach detach_callback; ///< Pointer to program info provider specific
                                                              callback to be invoked when a client detaches.
```

Use the same comment style for all fields rather than mixing styles.

Reply:
Updated.

Reply:
Had to push another commit due to build break. Can you please re-approve.